plugins/woocommerce/client/admin/client/settings-payments/offline/settings-payments-cod.tsx

```diff
@@ -0,0 +1,250 @@
+				createSuccessNotice(
+					__( 'Settings updated successfully', 'woocommerce' )
+				);
+			} )
+			.catch( () => {
+				createErrorNotice(
+					__( 'Failed to update settings', 'woocommerce' )
+				);
+			} );
+	};
+
+	return (
+		<PaymentSettingsLayout>
+			<GatewaySettingsForm>
+				<PaymentSettingsSection
+					title={ __( 'Enable and customise', 'woocommerce' ) }
+					description={ __(
+						'Choose how you want to present cash on delivery payments to your customers during checkout.',
+						'woocommerce'
+					) }
+				>
+					{ isLoading ? (
+						<FieldPlaceholder size="small" />
+					) : (
+						<CheckboxControl
+							label={ __(
+								'Enable cash on delivery payments',
+								'woocommerce'
+							) }
+							checked={ Boolean( formValues.enabled ) }
+							onChange={ ( checked ) => {
+								setFormValues( {
+									...formValues,
+									enabled: checked,
+								} );
+							} }
+						/>
+					) }
+					{ isLoading ? (
+						<FieldPlaceholder size="medium" />
+					) : (
+						<TextControl
+							label={ __( 'Title', 'woocommerce' ) }
+							help={ __(
+								'This controls the title which the user sees during checkout.',
+								'woocommerce'
+							) }
+							placeholder={ __(
+								'Cash on delivery payments',
+								'woocommerce'
+							) }
+							value={ String( formValues.title ) }
+							onChange={ ( value ) => {
+								setFormValues( {
+									...formValues,
+									title: value,
+								} );
+							} }
+						/>
+					) }
+					{ isLoading ? (
+						<FieldPlaceholder size="large" />
+					) : (
+						<TextareaControl
+							label={ __( 'Description', 'woocommerce' ) }
+							help={ __(
+								'Payment method description that the customer will see on your checkout.',
+								'woocommerce'
+							) }
+							value={ String( formValues.description ) }
+							onChange={ ( value ) => {
+								setFormValues( {
+									...formValues,
+									description: value,
+								} );
+							} }
+						/>
+					) }
+					{ isLoading ? (
+						<FieldPlaceholder size="large" />
+					) : (
+						<TextareaControl
+							label={ __( 'Instructions', 'woocommerce' ) }
+							help={ __(
+								'Instructions that will be added to the thank you page and emails.',
+								'woocommerce'
+							) }
+							value={ String( formValues.instructions ) }
+							onChange={ ( value ) => {
+								setFormValues( {
+									...formValues,
+									instructions: value,
+								} );
+							} }
+						/>
+					) }
+					{ isLoading || ! codSettings ? (
+						<FieldPlaceholder size="medium" />
+					) : (
+						<TreeSelectControl
+							label={ __(
+								'Enable for shipping methods',
+								'woocommerce'
+							) }
+							help={ __(
+								'Select shipping methods for which this payment method is enabled.',
+								'woocommerce'
+							) }
+							options={
+								codSettings.settings.enable_for_methods?.options
```

Comment:
this does not work locally 🐛  idea: maybe move it into useEffect and use ready data here?